File: FizzBuzz/java/src/tests/UnitTests.java
package tests;

import static org.junit.Assert.*;

import org.junit.Test;

import fizzbuzz.Solution;

public class UnitTests {

	@Test
	public void testSolution() {
		int n = 15;
        
        Solution sln = new Solution();
        String[] result = sln.solution(n);
        
        assertEquals("1", result[0]);
        assertEquals("2", result[1]);
        assertEquals("Fizz", result[2]);
        assertEquals("4", result[3]);
        assertEquals("Buzz", result[4]);
        // ...
        assertEquals("FizzBuzz", result[14]);
	}

}

File: FizzBuzz/csharp/UnitTests/UnitTests.cs
using System.Linq;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using FizzBuzz;

namespace UnitTests
{
    [TestClass]
    public class UnitTests
    {
        [TestMethod]
        public void Test_Solution()
        {
            var n = 15;

            var sln = new Solution();
            var result = sln.solution(n);

            Assert.AreEqual("1", result[0]);
            Assert.AreEqual("2", result[1]);            
            Assert.AreEqual("Fizz", result[2]);
            Assert.AreEqual("4", result[3]);
            Assert.AreEqual("Buzz", result[4]);
            // ...
            Assert.AreEqual("FizzBuzz", result[14]);
        }
    }
}
